app/src/main/java/com/example/training_musicplayer/ui/MainActivity.kt

```diff
@@ -0,0 +1,156 @@
+        updateSeekTime()
+        var a: String = ""
+        var b: String? = ""
+    }
+
+    private fun setAnimationDisk() {
+        anim = ObjectAnimator.ofFloat(
+            binding.imageMainRound,
+            View.ROTATION,
+            AppConstants.ROTATE_START,
+            AppConstants.ROTATE_END
+        )
+            .setDuration(AppConstants.SPEED_DISK.toLong())
+        anim.repeatCount = Animation.INFINITE
+        anim.interpolator = LinearInterpolator()
+        anim.start()
+    }
+
+    private fun updateSeekTime() {
+        binding.seekMainTime.progress = AppConstants.PROGRESS_INIT
+        binding.seekMainTime.max = mediaPlayer.duration
+        binding.textMainTimeEnd.text = Common.getTimeFormat(mediaPlayer.duration)
+        //Event SeekBar
+        binding.seekMainTime.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
+            override fun onProgressChanged(p0: SeekBar?, p1: Int, changed: Boolean) {
+                if (changed) {
+                    mediaPlayer.seekTo(p1)
+                }
+            }
+
+            override fun onStartTrackingTouch(p0: SeekBar?) {
+            }
+
+            override fun onStopTrackingTouch(p0: SeekBar?) {
+            }
+
+        })
+        runnable = Runnable {
+            binding.seekMainTime.progress = mediaPlayer.currentPosition
+            binding.textMainTimeStart.text = Common.getTimeFormat(mediaPlayer.currentPosition)
+            handler.postDelayed(runnable, AppConstants.TIME_DELAYED.toLong())
+        }
+        handler.postDelayed(runnable, AppConstants.TIME_DELAYED.toLong())
+        mediaPlayer.setOnCompletionListener {
+            binding.seekMainTime.progress = AppConstants.PROGRESS_INIT
+        }
+    }
+
+    private fun loadDataSong() {
+        nameSong = intent?.extras?.getString(AppConstants.EXTRA_NAME_SONG).orEmpty()
+        author = intent?.extras?.getString(AppConstants.EXTRA_AUTHOR).orEmpty()
+        url = intent?.extras?.getString(AppConstants.EXTRA_URL).orEmpty()
+        binding.textMainName.text = nameSong
+        binding.textMainAuthor.text = author
+        //
+        mediaPlayer = MediaPlayer()
+        mediaPlayer.setDataSource(url)
+        mediaPlayer.prepare()
+        mediaPlayer.start()
+        binding.textMainPlay.setBackgroundResource(R.drawable.ic_pause)
+
+    }
+
+    private fun setOnClicks() {
+        with(binding) {
+            controlButtons = listOf(
+                textMainBack,
+                textMainMore,
+                textMainReplay,
+                textMainPrev,
+                textMainPlay,
+                textMainNext,
+                textMainMenu
+            )
+        }
+        controlButtons.forEach { it.setOnClickListener(this) }
+    }
+
+    private fun initViewBinding() {
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+    }
+
+    @RequiresApi(Build.VERSION_CODES.KITKAT)
+    override fun onClick(p0: View?) {
```

Comment:
đặt tên biến rõ ràng ra em
```suggestion
    override fun onClick(view : View?) {
```

Reply:
em fix r a nhé